library/list.lisp

```diff
@@ -412,15 +432,15 @@
   (declare insertBy ((:a -> :a -> Ord) -> :a -> (List :a) -> (List :a)))
   (define (insertBy cmp x ys)
     "Generic version of insert"
-    (match ys
-      ((Nil)
-       (make-list x))
-      ((Cons y ys_)
-       (match (cmp x y)
-         ((GT)
-          (Cons y (insertBy cmp x ys_)))
-         (_
-          (Cons x ys))))))
+    (let ((rec 
+            (fn (ys acc)
+              (match ys
+                ((Nil) (Cons x acc))
+                ((Cons y yss)
+                 (match (cmp x y)
```

Comment:
why not just IF here?

Reply:
is `(if (== GT (cmp x y)) ... ...)` really better than a two-armed `match`? i'm ambivalent.

actually, this makes me think we should have a `(matches? PATTERN EXPR)` macro that returns `True` if the result of `EXPR` matches `PATTERN`, and `False` if it doesn't.

Reply:
Am I missing something? Why not `(if (> x y) ...)`?

Reply:
oh right. duh.

Reply:
The reason is that we don't actually have an `Ord` instance on `:a`, we only have a function which returns an `Ord` value. I don't really care whether we use an `if` with `==` or `match` in the body of the thing.

Reply:
ah, that makes sense. MATCH is fine then